### PR TITLE
Updates gpu macros

### PR DESCRIPTION
**Pull request by dipietrantonio**
- Removes duplicate error checking - gpu macros have built-in error checking.
- Removes outdated CMakeLists.txt files and build script.

### CODE REVIEW

File: src/gpu_macros.h [hunk 1]
```diff
@@ -88,10 +55,13 @@
 #define gpuEventCreate(...) GPU_CHECK_ERROR(cudaEventCreate(__VA_ARGS__))
 #define gpuGetDeviceCount(...) GPU_CHECK_ERROR(cudaGetDeviceCount(__VA_ARGS__))
 #define gpuGetLastError cudaGetLastError
+#define gpuGetDevice(...) GPU_CHECK_ERROR(cudaGetDevice(__VA_ARGS__))
+#define gpuSetDevice(...) GPU_CHECK_ERROR(cudaSetDevice(__VA_ARGS__))
+#define gpuDeviceGetAttribute(...) GPU_CHECK_ERROR(cudaDeviceGetAttribute(__VA_ARGS__))
+#define gpuDeviceAttributeWarpSize cudaDevAttrWarpSize
+#define __gpu_shfl_down(...) __shfl_down_sync(0xffffffff, __VA_ARGS__)
```
Comment (bwmeyers) [inline src/gpu_macros.h:62]: Note sure I understand these additions?

File: src/gpu_macros.h [hunk 2]
```diff
@@ -176,13 +122,25 @@
 #define gpuFloatComplex  hipFloatComplex
 #define make_gpuDoubleComplex make_hipDoubleComplex
 #define make_gpuFloatComplex make_hipFloatComplex
-
 #endif
 #define gpuCheckLastError(...) GPU_CHECK_ERROR(gpuGetLastError())
 #else
-// bool gpu_support() { return false;}
-// inline int num_available_gpus(){ return 0; } 
+inline int gpu_support() { return 0;}
```
Comment (bwmeyers) [inline src/gpu_macros.h:128]: Wait, wasn't this defined already earlier?

**Review by bwmeyers — CHANGES_REQUESTED** [review]
I think trying to keep the comments in the CMake files as much as possible is a good idea. Any particular reason why they were removed? I think it's just an issue of needing to merge the `main_gpu_bwm` branch back into `cdp-update-cmake` branch first?

Couple of other comments (probably just me misunderstanding things).

File: src/gpu_macros.h (hunk 1 above)
Comment (dipietrantonio) [reply src/gpu_macros.h:62]: Hi @bwmeyers, I have merged `main_gpu_bwm` into this branch. And fixed a comment that went lost in the way.

Those macros come from another project where I took the updated macros list. These are not used now, but may be in the future. There is no harm in having them there.

File: src/gpu_macros.h (hunk 2 above)
Comment (dipietrantonio) [reply src/gpu_macros.h:128]: They were commented out. I brought them back with adjustments for the current code.

**Review by bwmeyers — APPROVED** [review]

**Pull request merged**